code/modules/paperwork/photography.dm

```diff
@@ -38,6 +38,14 @@
 	var/sillynewscastervar  //Photo objects with this set to 1 will not be ejected by a newscaster. Only gets set to 1 if a silicon puts one of their images into a newscaster
 
 
+/obj/item/weapon/photo/Destroy()
+	if(istype(loc, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A = loc
+		if(A.note == src)
+			A.note = null

```

Comment:
Instead of these overrides, add a `handle_atom_del()` to the airlock that checks if the atom deleted is its note and sets `note = null` if it is.